<launch>
  <rosparam command="load" file="$(find viscon)/config/tello_topics.yaml" />
  <arg name="tello_ip" default="192.168.10.1" />
    <arg name="tello_cmd_server_port" default="8889" />
    <arg name="local_cmd_client_port" default="8890" />
    <arg name="local_vid_server_port" default="6038" />
    <arg name="namespace" default="tello" />

    <group ns="$(arg namespace)">
        <node pkg="tello_driver" name="tello" type="tello_driver_node.py" output="screen">
            <param name="local_cmd_client_port" value="$(arg local_cmd_client_port)" />
            <param name="local_vid_server_port" value="$(arg local_vid_server_port)" />
            <param name="tello_ip" value="$(arg tello_ip)" />
            <param name="tello_cmd_server_port" value="$(arg tello_cmd_server_port)" />
            <param name="connect_timeout_sec" value="10.0" />
            <param name="stream_h264_video" value="false" />
        </node>

        <node pkg="image_transport" name="image_compressed" type="republish" args="raw in:=image_raw compressed out:=image_raw" />
    </group>
  <dynamic_reconfigure command="dynparam load" node="/tello/tello" file="$(find viscon)/config/dynamic_params.yaml" />

  <node name="orb_slam2_mono" pkg="orb_slam2_ros"
      type="orb_slam2_ros_mono" output="screen">

       <remap from="/camera/image_raw" to="/tello/image_raw" />

       <param name="publish_pointcloud" type="bool" value="true" />
       <param name="publish_pose" type="bool" value="true" />
       <param name="localize_only" type="bool" value="false" />
       <param name="reset_map" type="bool" value="false" />
 
       <!-- static parameters -->
       <param name="load_map" type="bool" value="false" />
       <!--param name="map_file" type="string" value="map.bin" /-->
       <param name="map_file" type="string" value="$(find orb_slam2_ros)/ros/config/map.bin" /> 
       <param name="settings_file" type="string" value="$(find orb_slam2_ros)/orb_slam2/config/Tello.yaml" />
       <param name="voc_file" type="string" value="$(find orb_slam2_ros)/orb_slam2/Vocabulary/ORBvoc.txt" />

       <param name="pointcloud_frame_id" type="string" value="map" />
       <param name="camera_frame_id" type="string" value="camera_link" />
       <param name="min_num_kf_in_map" type="int" value="5" />
  </node>
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find orb_slam2_ros)/ros/config/rviz_config.rviz" />
  <node type="rqt_reconfigure" name="dynamic_reconfigure" pkg="rqt_reconfigure"/>
  <node type="teleop_twist_keyboard.py" name="twist" pkg="drone_control" launch-prefix="xterm -e" output="screen"/>
  <node pkg="viscon" name="visual_simple_pid" type="slam_control_topic.py" output="screen" launch-prefix="xterm -e"/>
  <node pkg="viscon" name="head" type="head.py" output="screen" launch-prefix="xterm -e"/>
</launch>
